src/parser/datetime.rs

```diff
@@ -104,14 +194,112 @@ fn potential_interval_tokens(from: &DateTimeField) -> Vec<IntervalToken> {
+fn build_timezone_offset_second(
+    tokens: &Vec<IntervalToken>,
+    value: &str,
+) -> Result<i64, ParserError> {
+    use IntervalToken::*;
+    let all_formats = [
+        vec![Plus, Num(0), Colon, Num(0)],
+        vec![Dash, Num(0), Colon, Num(0)],
+        vec![Plus, Num(0), Num(0)],
+        vec![Dash, Num(0), Num(0)],
+        vec![Plus, Num(0)],
+        vec![Dash, Num(0)],
+        vec![Name("".to_string())],
+        vec![Zulu],
+    ];
+
+    let mut is_positive = true;
+    let mut hour_offset: i64 = -1;
+    let mut minute_offset: i64 = -1;
+
+    for format in all_formats.iter() {
+        let expected = format.to_vec();
+        let actual = tokens.iter();
+
+        if actual.len() != expected.len() {
+            continue;
+        }
+
+        for (i, (atok, etok)) in actual.zip(&expected).enumerate() {
+            match (atok, etok) {
+                (Colon, Colon) | (Plus, Plus) => { /* Matching punctuation */ }
+                (Dash, Dash) => {
+                    is_positive = false;
+                }
+                (Num(val), Num(_)) => {
+                    let val = *val;
+                    if hour_offset == -1 {
+                        // We haven't seen an hour offset yet so lets try to parse that
+                        if val <= 24 {
+                            hour_offset = val.try_into().unwrap();
+                        } else {
+                            return Err(ParserError::ParserError(format!(
+                                "Error parsing timezone string ({}): timezone hour invalid {}",
+                                value, val
+                            )));
+                        }
+                    } else if minute_offset == -1 {
+                        if val <= 60 {
+                            minute_offset = val.try_into().unwrap();
+                        } else {
+                            return Err(ParserError::ParserError(format!(
+                                "Error parsing timezone string ({}): timezone minute invalid {}",
+                                value, val
+                            )));
+                        }
+                    } else {
+                        // We've already seen an hour and a minute so we should never see another number
+                        return Err(ParserError::ParserError(format!(
+                            "Error parsing timezone string ({}): invalid value {} at token index {}", value,
+                            val, i
+                        )));
+                    }
+                }
+                (Zulu, Zulu) => return Ok(0 as i64),
+                (Name(val), Name(_)) => {
+                    // For now, we don't support named timezones
+                    return Err(ParserError::ParserError(format!(
+                            "Error parsing timezone string ({}): named timezones are not supported. Failed to parse {} at token index {}", value, val, i)));
```

Comment:
could you wrap this string? newlines in string literals are allowed in rust, and `\\` at the end of a line removes the newline and all leading whitespace on the next line, so this would be equivalent:

```
                    return Err(ParserError::ParserError(format!(
                        "Error parsing timezone string ({}): named timezones \
                        are not supported. Failed to parse {} \
                        at token index {}", value, val, i)));
 ```